stock_storage_type/models/stock_location.py

```diff
@@ -55,6 +54,81 @@ class StockLocation(models.Model):
         "location_id",
         string="Storage locations sequences",
     )
+    location_is_empty = fields.Boolean(
+        compute="_compute_location_is_empty",
+        store=True,
+        help="technical field: True if the location is empty "
+        "and there is no pending incoming products in the location"
+    )
+
+    in_move_ids = fields.One2many(
+        'stock.move', 'location_dest_id',
+        domain=[('state', 'not in', ('draft', 'done', 'cancel'))],
+        help="technical field: the pending incoming stock.moves in the location",
+    )
+    in_move_line_ids = fields.One2many(
+        'stock.move.line', 'location_dest_id',
+        domain=[('state', 'not in', ('draft', 'done', 'cancel'))],
+        help="technical field: the pending incoming "
+        "stock.move.lines in the location",
+    )
+    children_in_move_line_ids = fields.Many2many(
+        'stock.move.line',
+        _compute='_compute_children_in_move_line_ids',
+        store=True,
+        help="technical field: the pending incoming stock.move.lines "
+        "in the location or children locations",
+    )
+    location_will_contain_lot_ids = fields.Many2many(
+        'stock.production.lot',
+        store=True,
+        compute="_compute_location_will_contain_lot_ids",
+        help="technical field: list of stock.production.lots in "
+        "the location, either now or in pending operations"
+    )
+    location_will_contain_product_ids = fields.Many2many(
+        'product.product',
+        store=True,
+        compute="_compute_location_will_contain_product_ids",
+        help="technical field: list of products in "
+        "the location, either now or in pending operations"
+    )
+
+    @api.depends('child_ids.in_move_line_ids')
+    def _compute_children_in_move_line_ids(self):
+        for rec in self:
+            rec.children_in_move_line_ids = (
+                rec.mapped('child_ids.in_move_line_ids') | rec.in_move_line_ids
```

Comment:
It will contain only the lines of the current location and the direct children, not more than one level, is it what we expect?

Reply:
nope, you are correct this was a late change when I noticed the search on stock.move.lines was different from the one on stock.move

Reply:
should be fixed now